apps/barista-design-system/src/pages/single-page/components/toc/toc.ts

```diff
@@ -87,23 +86,13 @@ export class BaToc implements OnInit, AfterViewInit, OnDestroy {
   }
 
   /** @internal handle the click on a TOC item */
-  _handleTocClick(ev: MouseEvent): void {
-    /* Preventing the default behavior is necessary, because on Angular component pages
-     * there's a base URL defined and the on-page-links are always relative to "/"
-     * and not to the current page. */
-    ev.preventDefault();
-    ev.stopImmediatePropagation();
-    const targetId = (ev.currentTarget as HTMLElement).getAttribute('href');
-    const target = this._document.querySelector(targetId || '');
-
-    if (this._platform.isBrowser && target) {
-      // Has to be set manually because of preventDefault() above.
-      window.location.hash = targetId || '';
-      requestAnimationFrame(() => {
-        target.scrollIntoView({
-          behavior: 'smooth',
-        });
+  _handleTocClick(target: string): void {
+    this._location.go(`${this._location.path()}#${target}`);
+    const targetHeadline = this._document.querySelector(`#${target}`);
```

Comment:
NIT: I know this should not happen, but with this selector we could potentially limit id clashes that may occur within the document. 
```suggestion
    const targetHeadline = this._document.querySelector(`h2[id="${target}"], h3[id="${target}"]`);
```